src/shims/foreign_items.rs

```diff
@@ -202,6 +202,13 @@ trait EvalContextExtPriv<'tcx>: crate::MiriInterpCxExt<'tcx> {
         if !align.is_power_of_two() {
             throw_ub_format!("creating allocation with non-power-of-two alignment {}", align);
         }
+        if let Err(AlignFromBytesError::TooLarge(_)) = Align::from_bytes(align) {
```

Comment:
Good idea to call `from_bytes`. Now you can even do the power-of-two checking that way, removing the manual check above. :)